cpp/memilio/math/interpolation.h

```diff
@@ -34,21 +37,51 @@ namespace mio
  * @param[in] x_2 Right node of interpolation.
  * @param[in] y_1 Value at left node.
  * @param[in] y_2 Value at right node.
- * @param[out] unnamed Interpolation result.
+ * @return Interpolation result.
  */
 template <typename X, typename V>
 auto linear_interpolation(const X& x_eval, const X& x_1, const X& x_2, const V& y1, const V& y2)
 {
-    auto weight = (x_eval - x_1) / (x_2 - x_1);
+    const auto weight = (x_eval - x_1) / (x_2 - x_1);
     return y1 + weight * (y2 - y1);
 }
 
+/**
+ * @brief Linear interpolation of a TimeSeries.
+ * Assumes that the time points are monotonic increasing. If the time series is strictly monotonic, this function is
+ * continuous in time.

```

Comment:
Strictly speaking, I did not say that ;)
But, anyways, I use std::upper_bound in the implementation, which requires that the data (i.e. time points) is ordered by std::less, hence strictly monotonic. It still works if it is not, but take the TimeSeries
time | 0 | 1 | 1 | 2
---|---|---|---|---
value | 0 | 1 | 0 | 1

for example.

Since std::upper_bound returns the _first_ time point with a time strictly after a given interpolation time, we get the function x in (0, 1) and x-1 in [1, 2), because linear_interpolation  uses the time point one before the bound as the previous time point. These previous time points are either (0, 0) or (1, 0) from the table. This results in a kind of sawtooth function `../|/''` with a discontinuity at 1.

If instead of (1,0) we use e.g. (0.5, x), the results should the same in [0, 1), but the interval [1, 2) will be stretched, I think.